ke/compiler.py

```diff
@@ -214,6 +214,20 @@ def compile_either(e, macros):
             elif isinstance(c, asm.CharacterClass):
                 characters += c.characters
         return asm.CharacterClass(characters, False)
+    
```

Comment:
this complex logic deserves to be in its own function, with a nice self-documenting name and some good unit tests :)